examples/threading/threading.c

```diff
@@ -10,7 +13,35 @@
 
 void* threadfunc(void* thread_param)
 {
-
+    int usleep_result;
+    int mx_result;
+    
+    struct thread_data* thread_func_args = (struct thread_data*)thread_param;
+    usleep_result = usleep(thread_func_args->wait_to_obtain_ms * 1000);
+    if(usleep_result == -1)
+    {
+        ERROR_LOG("\nusleep() execution failed. Value returned is %d",usleep_result);
+        perror("Error:");
+    }
+    mx_result = pthread_mutex_lock(thread_func_args->mutex);
+    if(mx_result)
+    {
+        errno = mx_result;
+        ERROR_LOG("\nMutex could not be locked. Error: %s", strerror(errno));
+    }
+    usleep_result = usleep(thread_func_args->wait_to_release_ms * 1000);
+    if(usleep_result == -1)
+    {
+        ERROR_LOG("\nusleep() execution failed. Value returned is %d",usleep_result);
+        perror("Error:");
+    }
+    mx_result = pthread_mutex_unlock(thread_func_args->mutex);
+    if(mx_result)
+    {
+        errno = mx_result;
+        ERROR_LOG("\nMutex could not be unlocked. Error: %s", strerror(mx_result));
+    }
+    thread_func_args->thread_complete_success = true;
```

Comment:
@darshitagrawal did the thread complete successfully if each of the steps above failed?

Reply:
Yes, it did.